Documentation/admin/logging-destination.md

```diff
@@ -127,8 +127,117 @@ Installing the x-pack plugin on your Elasticsearch nodes enables authentication
+You should then configure a policy for the ES cluster policy thus, with appropriate substitutions for the capitalized terms:
+ 
+```
+{
+  "Version": "2017-08-14",
+   "Statement": [
+     {
+       "Effect": "Allow",
+       "Principal": {
+         "AWS": "arn:aws:iam::ACCOUNT:role/ROLE"
+       },
+       "Action": "es:*",
+       "Resource": "arn:aws:es:us-east-1:ACCOUNT:domain/ES_DOMAIN/*"
+     },
+     {
+       "Effect": "Allow",
+       "Principal": {
+         "AWS": "*"
+       },
+       "Action": "es:*",
+       "Resource": "arn:aws:es:us-east-1:ACCOUNT:domain/ES_DOMAIN/*",
+       "Condition": {
+         "IpAddress": {
+           "aws:SourceIp": "XXX.XX.XXX.XX/32"
+         }
+       }
+     }
+   ]
+}
+ 
+```
+This will allow your fluentd hosts (by virtue of the possession of the role) and any traffic coming from the specified IP addresses (you querying Kibana) to access the various endpoints. Whilst not ideally secure (both the fluentd and Kibana boxes should ideally be restricted to the verbs they require) it should allow you to get up and ingesting logs without anything getting in your way, before you tighten down the policy.
+
+Additionally, you can use an STS assumed role as the authenticating factor and instruct the plugin to assume this role. This is useful for cross-account access and when assigning a standard role is not possible. The endpoint configuration looks like:
+
+ ```
+  <endpoint>
+     url https://CLUSTER_ENDPOINT_URL
+     region eu-east-1
+     assume_role_arn arn:aws:sts::ACCOUNT:assumed-role/ROLE
+     assume_role_session_name SESSION_ID # Defaults to fluentd if omitted
+  </endpoint>
+   
+   ```
+ 
+ The policy attached to your AWS Elasticsearch cluster then becomes something like:
+ 
+ ```
+ {
+   "Version": "2017-08-15",
+   "Statement": [
+     {
+       "Effect": "Allow",
+       "Principal": {
+         "AWS": "arn:aws:sts::ACCOUNT:assumed-role/ROLE/SESSION_ID"
+       },
+       "Action": "es:*",
+       "Resource": "arn:aws:es:eu-east-1:ACCOUNT:domain/ES_DOMAIN/*"
+     }
+   ]
+ }
+ 
+ ```
+ 
+You'll need to ensure that the environment in which the fluentd plugin runs has the capability to assume this role, by attaching a policy something like this to the instance profile:
+ 
+ ```
+ {
+     "Version": "2017-08-15",
+     "Statement": {
+         "Effect": "Allow",
+         "Action": "sts:AssumeRole",
+         "Resource": "arn:aws:iam::ACCOUNT:role/ROLE"
+     }
+ }
+ 
+ ```
+ 
+Next, we're going to configure Fluentd to gather the logs that we're interested in looking at and deliver them to our brand  new ElasticSearch machine. All of this is configured through the [td-agent configuration file](https://docs.treasuredata.com/articles/td-agent). We're going to add the below configuration in the Kubernetes repository for the [fluentd plugin](https://github.com/kubernetes/kubernetes/blob/7e1b9dfd0fc75311ff6339f19b514e8caaebeafd/cluster/addons/fluentd-elasticsearch/fluentd-es-image/td-agent.conf).
```

Comment:
Next, configure fluentd to gather the selected logs and deliver them to the Elasticsearch machine. Use the  [td-agent configuration file][td-config] to configure fluentd. Add the following configuration to the Kubernetes repository for the [fluentd plugin][fluentd-plugin].

Reply:
Done!